Something went wrong in the testing here. First traceback, documenting here:

```
2025-04-08 18:59:12,855 WARNING  [-] dispatcherd.service.control_tasks PID:55613 Canceling task in worker 35, task: {'task': 'awx.main.tasks.jobs.RunJob', 'args': [260], 'uuid': 'a81d2fe3-c7fb-4144-9af9-4152c7ef143a'}
2025-04-08 18:59:12,856 ERROR    [-] dispatcherd.worker.task PID:88907 Worker 31 task canceled (uuid=045fac66-28d0-4c49-b810-01c2b3173911)
Traceback (most recent call last):
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/awx/main/tasks/jobs.py", line 412, in acquire_lock
    fcntl.lockf(self.lock_fd, fcntl.LOCK_EX | fcntl.LOCK_NB)
BlockingIOError: [Errno 11] Resource temporarily unavailable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/awx/main/tasks/jobs.py", line 499, in run
    self.build_project_dir(self.instance, private_data_dir)
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/awx/main/tasks/jobs.py", line 1107, in build_project_dir
    self.sync_and_copy(job.project, private_data_dir, scm_branch=job.scm_branch)
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/awx/main/tasks/jobs.py", line 777, in sync_and_copy
    self.acquire_lock(project, self.instance.id)
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/awx/main/tasks/jobs.py", line 423, in acquire_lock
    time.sleep(1.0)
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/dispatcherd/worker/task.py", line 34, in task_cancel
    raise DispatcherCancel
dispatcherd.worker.task.DispatcherCancel

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/dispatcherd/worker/task.py", line 126, in run_callable
    return _call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/awx/main/tasks/jobs.py", line 101, in _wrapped
    return f(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/awx/main/tasks/signals.py", line 81, in _wrapped
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/awx/main/tasks/jobs.py", line 634, in run
    self.runner_callback.delay_update(result_traceback=traceback.format_exc())
                                                       ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/traceback.py", line 184, in format_exc
    return "".join(format_exception(*sys.exc_info(), limit=limit, chain=chain))
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/traceback.py", line 139, in format_exception
    te = TracebackException(type(value), value, tb, limit=limit, compact=True)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/traceback.py", line 783, in __init__
    context = TracebackException(
              ^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/traceback.py", line 728, in __init__
    self.stack = StackSummary._extract_from_extended_frame_gen(
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/traceback.py", line 413, in _extract_from_extended_frame_gen
    for f, (lineno, end_lineno, colno, end_colno) in frame_gen:
  File "/usr/lib64/python3.11/traceback.py", line 350, in _walk_tb_with_full_positions
    positions = _get_code_position(tb.tb_frame.f_code, tb.tb_lasti)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/traceback.py", line 364, in _get_code_position
    return next(itertools.islice(positions_gen, instruction_index // 2, None))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/dispatcherd/worker/task.py", line 34, in task_cancel
    raise DispatcherCancel
dispatcherd.worker.task.DispatcherCancel
2025-04-08 18:59:12,868 ERROR    [-] dispatcherd.worker.task PID:90786 Worker failed to run task awx.main.scheduler.tasks.workflow_manager(*[], **{}
Traceback (most recent call last):
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/django/db/transaction.py", line 263, in __exit__
    connection.commit()
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/django/utils/asyncio.py", line 26, in inner
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/django/db/backends/base/base.py", line 337, in commit
    self._commit()
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/django/db/backends/base/base.py", line 313, in _commit
    return self.connection.commit()
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/psycopg/connection.py", line 256, in commit
    self.wait(self._commit_gen())
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/psycopg/connection.py", line 394, in wait
    return waiting.wait(gen, self.pgconn.socket, interval=interval)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/psycopg/waiting.py", line 346, in wait_poll
    fileevs = poll.poll(interval)
              ^^^^^^^^^^^^^^^^^^^
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/dispatcherd/worker/task.py", line 34, in task_cancel
    raise DispatcherCancel
dispatcherd.worker.task.DispatcherCancel

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/django/db/backends/postgresql/base.py", line 375, in _set_autocommit
    self.connection.autocommit = autocommit
    ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/psycopg/_connection_base.py", line 179, in autocommit
    self._set_autocommit(value)
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/psycopg/connection.py", line 407, in _set_autocommit
    self.set_autocommit(value)
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/psycopg/connection.py", line 412, in set_autocommit
    self.wait(self._set_autocommit_gen(value))
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/psycopg/connection.py", line 394, in wait
    return waiting.wait(gen, self.pgconn.socket, interval=interval)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/psycopg/waiting.py", line 335, in wait_poll
    s = next(gen)
        ^^^^^^^^^
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/psycopg/_connection_base.py", line 185, in _set_autocommit_gen
    yield from self._check_intrans_gen("autocommit")
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/psycopg/_connection_base.py", line 258, in _check_intrans_gen
    raise e.ProgrammingError(
psycopg.ProgrammingError: can't change 'autocommit' now: connection in transaction status ACTIVE

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/ansible_base/lib/utils/db.py", line 109, in django_pglocks_advisory_lock
    yield acquired
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/ansible_base/lib/utils/db.py", line 143, in advisory_lock
    yield internal_lock
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/awx/main/scheduler/task_manager.py", line 149, in schedule
    with transaction.atomic():
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/django/db/transaction.py", line 307, in __exit__
    connection.set_autocommit(True)
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/django/db/backends/base/base.py", line 494, in set_autocommit
    self._set_autocommit(autocommit)
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/django/db/backends/postgresql/base.py", line 374, in _set_autocommit
    with self.wrap_database_errors:
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/django/db/utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/django/db/backends/postgresql/base.py", line 375, in _set_autocommit
    self.connection.autocommit = autocommit
    ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/psycopg/_connection_base.py", line 179, in autocommit
    self._set_autocommit(value)
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/psycopg/connection.py", line 407, in _set_autocommit
    self.set_autocommit(value)
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/psycopg/connection.py", line 412, in set_autocommit
    self.wait(self._set_autocommit_gen(value))
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/psycopg/connection.py", line 394, in wait
    return waiting.wait(gen, self.pgconn.socket, interval=interval)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/psycopg/waiting.py", line 335, in wait_poll
    s = next(gen)
        ^^^^^^^^^
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/psycopg/_connection_base.py", line 185, in _set_autocommit_gen
    yield from self._check_intrans_gen("autocommit")
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/psycopg/_connection_base.py", line 258, in _check_intrans_gen
    raise e.ProgrammingError(
django.db.utils.ProgrammingError: can't change 'autocommit' now: connection in transaction status ACTIVE

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/django/db/backends/utils.py", line 87, in _execute
    return self.cursor.execute(sql)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/psycopg/cursor.py", line 97, in execute
    raise ex.with_traceback(None)
psycopg.OperationalError: sending query failed: another command is already in progress

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/dispatcherd/worker/task.py", line 156, in perform_work
    result = self.run_callable(message)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/dispatcherd/worker/task.py", line 126, in run_callable
    return _call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/awx/main/scheduler/tasks.py", line 35, in workflow_manager
    run_manager(WorkflowManager, "workflow")
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/awx/main/scheduler/tasks.py", line 20, in run_manager
    manager().schedule()
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/awx/main/scheduler/task_manager.py", line 148, in schedule
    with advisory_lock(f"{self.prefix}_lock", lock_session_timeout_milliseconds=lock_session_timeout_milliseconds, wait=False) as acquired:
  File "/usr/lib64/python3.11/contextlib.py", line 158, in __exit__
    self.gen.throw(typ, value, traceback)
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/ansible_base/lib/utils/db.py", line 142, in advisory_lock
    with django_pglocks_advisory_lock(*args, **kwargs) as internal_lock:
  File "/usr/lib64/python3.11/contextlib.py", line 158, in __exit__
    self.gen.throw(typ, value, traceback)
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/ansible_base/lib/utils/db.py", line 115, in django_pglocks_advisory_lock
    cursor.execute(command)
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/django/db/backends/utils.py", line 67, in execute
    return self._execute_with_wrappers(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/django/db/backends/utils.py", line 80, in _execute_with_wrappers
    return executor(sql, params, many, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/django/db/backends/utils.py", line 84, in _execute
    with self.db.wrap_database_errors:
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/django/db/utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/django/db/backends/utils.py", line 87, in _execute
    return self.cursor.execute(sql)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/psycopg/cursor.py", line 97, in execute
    raise ex.with_traceback(None)
django.db.utils.OperationalError: sending query failed: another command is already in progress
```

Seems that something might have went wrong in the process of canceling a task.

After that, we seem to consistently get this error in the worker and it fails to recover:

```
2025-04-08 18:59:13,726 ERROR    [-] dispatcherd.worker.task PID:90786 Worker failed to run task awx.main.scheduler.tasks.workflow_manager(*[], **{}
Traceback (most recent call last):
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/django/db/backends/utils.py", line 87, in _execute
    return self.cursor.execute(sql)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/psycopg/cursor.py", line 97, in execute
    raise ex.with_traceback(None)
psycopg.OperationalError: sending query failed: another command is already in progress

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/dispatcherd/worker/task.py", line 156, in perform_work
    result = self.run_callable(message)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/dispatcherd/worker/task.py", line 126, in run_callable
    return _call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/awx/main/scheduler/tasks.py", line 35, in workflow_manager
    run_manager(WorkflowManager, "workflow")
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/awx/main/scheduler/tasks.py", line 20, in run_manager
    manager().schedule()
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/awx/main/scheduler/task_manager.py", line 148, in schedule
    with advisory_lock(f"{self.prefix}_lock", lock_session_timeout_milliseconds=lock_session_timeout_milliseconds, wait=False) as acquired:
  File "/usr/lib64/python3.11/contextlib.py", line 137, in __enter__
    return next(self.gen)
           ^^^^^^^^^^^^^^
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/ansible_base/lib/utils/db.py", line 138, in advisory_lock
    idle_in_transaction_session_timeout = cur.execute("SHOW idle_in_transaction_session_timeout").fetchone()[0]
                                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/django/db/backends/utils.py", line 67, in execute
    return self._execute_with_wrappers(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/django/db/backends/utils.py", line 80, in _execute_with_wrappers
    return executor(sql, params, many, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/django/db/backends/utils.py", line 84, in _execute
    with self.db.wrap_database_errors:
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/django/db/utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/django/db/backends/utils.py", line 87, in _execute
    return self.cursor.execute(sql)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/lib/awx/venv/awx/lib64/python3.11/site-packages/psycopg/cursor.py", line 97, in execute
    raise ex.with_traceback(None)
django.db.utils.OperationalError: sending query failed: another command is already in progress
```

--- a/awx/main/tasks/jobs.py
+++ b/awx/main/tasks/jobs.py
@@ -17,6 +17,7 @@ import urllib.parse as urlparse
 
 # Django
 from django.conf import settings
+from django.db import transaction
 
 # Shared code for the AWX platform
 from awx_plugins.interfaces._temporary_private_container_api import CONTAINER_ROOT, get_incontainer_path
@@ -40,7 +41,6 @@ from awx.main.constants import (
     JOB_FOLDER_PREFIX,
     MAX_ISOLATED_PATH_COLON_DELIMITER,
     CONTAINER_VOLUMES_MOUNT_TYPES,
-    ACTIVE_STATES,
     HOST_FACTS_FIELDS,
 )
 from awx.main.models import (
@@ -115,7 +115,7 @@ def with_path_cleanup(f):
 
 @task(on_duplicate='queue_one', bind=True)
 def dispatch_waiting_jobs(binder):
-    for uj in UnifiedJob.objects.filter(status='waiting', controller_node=settings.CLUSTER_HOST_ID).only('id', 'status', 'polymorphic_ctype'):
+    for uj in UnifiedJob.objects.filter(status='waiting', controller_node=settings.CLUSTER_HOST_ID).only('id', 'status', 'polymorphic_ctype', 'celery_task_id'):
         binder.control('run', data={'task': serialize_task(uj._get_task_class()), 'args': [uj.id], 'uuid': uj.celery_task_id})
 
 
@@ -126,6 +126,7 @@ class BaseTask(object):
     callback_class = RunnerCallback
 
     def __init__(self):
+        self.instance = None
         self.cleanup_paths = []
         self.update_attempts = int(getattr(settings, 'DISPATCHER_DB_DOWNTOWN_TOLLERANCE', settings.DISPATCHER_DB_DOWNTIME_TOLERANCE) / 5)
         self.runner_callback = self.callback_class(model=self.model)
@@ -466,12 +467,20 @@ class BaseTask(object):
         """
         Run the job/task and capture its output.
         """
-        self.instance = self.model.objects.get(pk=pk)
-        if self.instance.status != 'canceled' and self.instance.cancel_flag:
-            self.instance = self.update_model(self.instance.pk, start_args='', status='canceled')
-        if self.instance.status not in ACTIVE_STATES:
-            # Prevent starting the job if it has been reaped or handled by another process.
-            raise RuntimeError(f'Not starting {self.instance.status} task pk={pk} because {self.instance.status} is not a valid active state')
+        if not self.instance:  # Used to skip fetch for local runs
+            with transaction.atomic():
+                self.instance = self.model.objects.select_for_update().get(pk=pk)
+                if self.instance.cancel_flag and self.instance.status != 'canceled':
+                    self.instance.start_args = ''
+                    self.instance.status = 'canceled'
+                    self.instance.save(update_fields=['start_args', 'status'])
+                if self.instance.status != 'waiting':
+                    # Prevent starting the job if it has been reaped or had a duplicate task.
+                    raise RuntimeError(f'Not starting {self.instance.status} task pk={pk} because its status "{self.instance.status}" is not "waiting"')
+                # self.instance because of the update_model pattern and when it's used in callback handlers
+                self.instance.status = 'running'
+                self.instance.start_args = ''  # blank field to remove encrypted passwords
+                self.instance.save(update_fields=['start_args', 'status'])
 
         if self.instance.execution_environment_id is None:
             from awx.main.signals import disable_activity_stream
@@ -479,8 +488,6 @@ class BaseTask(object):
             with disable_activity_stream():
                 self.instance = self.update_model(self.instance.pk, execution_environment=self.instance.resolve_execution_environment())
 
-        # self.instance because of the update_model pattern and when it's used in callback handlers
-        self.instance = self.update_model(pk, status='running', start_args='')  # blank field to remove encrypted passwords
         self.instance.websocket_emit_status("running")
         status, rc = 'error', None
         self.runner_callback.event_ct = 0
@@ -748,6 +755,7 @@ class SourceControlMixin(BaseTask):
             try:
                 # the job private_data_dir is passed so sync can download roles and collections there
                 sync_task = RunProjectUpdate(job_private_data_dir=private_data_dir)
+                sync_task.instance = local_project_sync  # avoids "waiting" status check, performance
                 sync_task.run(local_project_sync.id)
                 local_project_sync.refresh_from_db()
                 self.instance = self.update_model(self.instance.pk, scm_revision=local_project_sync.scm_revision)

--- a/awx/settings/production_defaults.py
+++ b/awx/settings/production_defaults.py
@@ -4,6 +4,7 @@
 # Production settings for AWX project.
 
 import os
+from pathlib import Path
 
 DEBUG = False
 TEMPLATE_DEBUG = DEBUG
@@ -23,8 +24,14 @@ ALLOWED_HOSTS = []
 # only used for deprecated fields and management commands for them
 BASE_VENV_PATH = os.path.realpath("/var/lib/awx/venv")
 
+# Switch to a writable location for the dispatcher sockfile location
+DISPATCHERD_DEBUGGING_SOCKFILE = str(Path().home() / 'dispatcherd.sock')
+
 # Very important that this is editable (not read_only) in the API
 AWX_ISOLATION_SHOW_PATHS = [
     '/etc/pki/ca-trust:/etc/pki/ca-trust:O',
     '/usr/share/pki:/usr/share/pki:O',
 ]
+
+del os
+del Path

--- a/requirements/requirements.txt
+++ b/requirements/requirements.txt
@@ -356,7 +356,7 @@ protobuf==5.29.3
     #   opentelemetry-proto
 psutil==6.1.1
     # via -r /awx_devel/requirements/requirements.in
-psycopg==3.2.3
+psycopg==3.2.6
     # via -r /awx_devel/requirements/requirements.in
 ptyprocess==0.7.0
     # via pexpect

